lib/rig_inbound_gateway_web/v1/metadata_controller.ex

```diff
@@ -0,0 +1,195 @@
+
+      userid -> connection token
+
+      conection token ->
+        userid
+        locale
+        timezone
+
+  All indexed fields are then being propagated to all other instances, so they know where to find the corresponding metadata.
+
+  ## Example
+
+  A user connects to RIG 1 and sends the metadata fields from above to RIG 1. RIG 1 would propagate the connection token and the userid along with the RIG instance ID to all other RIG instances.
+  If someone were to access the metadata of a user from RIG 3, RIG 3 would know where to find the metadata, ask RIG 1 for it and return the metadata set.
+
+  ## Dirty testing with JWT
+
+      AUTH='Authorization:Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI5YWIxYmZmMi1hOGQ4LTQ1NWMtYjQ4YS01MDE0NWQ3ZDhlMzAiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1Njg3MTMzNjEsImV4cCI6NDEwMzI1ODE0M30.kjiR7kFyOEeMJaY1zPCctut39eEWmKswUCNZdK5Q3-w'
+      META='{ "metadata": { "locale": "de-AT", "timezone": "GMT+2" } }'
+      http put ":4000/_rig/v1/connection/sse/${CONN_TOKEN}/metadata" <<< "$META" "$AUTH"
+
+  ## Dirty testing without JWT
+      META='{ "metadata": { "userid": "9ab1bff2-a8d8-455c-b48a-50145d7d8e30", "locale": "de-AT", "timezone": "GMT+2" } }'
+      http put ":4000/_rig/v1/connection/sse/${CONN_TOKEN}/metadata" <<< "$META"
+
+  """
+  @spec set_metadata(conn :: Plug.Conn.t(), params :: map) :: Plug.Conn.t()
+  def set_metadata(
+        %{method: "PUT"} = conn,
+        %{
+          "connection_id" => connection_id
+        }
+      ) do
+
+    {err, data} = conn
+    |> accept_only_req_for(["application/json"])
+    |> extract_metadata()
+
+    if err === :error do
+      fail!(conn, data)
+    else
+      persist_metadata(data, connection_id)
+      send_resp(conn, :no_content, "")
+    end
+  end
+
+  # ---
+
+  defp extract_metadata(conn) do
+    {json_metadata_err, json_metadata} = extract_metadata_from_json(conn)
+    jwt_metadata = extract_metadata_from_jwt(conn)
+
+    if json_metadata_err === :error do
+      {:error, json_metadata}
+    else
+      # Merge values from metadata with the values from JWT, prioritizing JWT values
+      metadata = Map.merge(json_metadata, jwt_metadata)
+
+      # Check if metadata contains all the indexed fields; if not: send back a bad request
+      # This needs to be done here because now the values from the JWT got inserted
+      if has_required_indices?(metadata) do
+        {:ok, metadata}
+      else
+        message = """
+        Metadata doesn't contain indexed fields.
+        """
+        {:error, message}
+      end
+    end
+  end
+
+  # ---
+
+  defp extract_metadata_from_jwt(conn) do
+    with {:ok, claims} <- Map.get(conn.assigns, :auth_tokens, [])
+    |> Enum.map(fn
+      {"bearer", token} -> JWT.parse_token(token)
+      _ -> {:ok, %{}}
+    end)
+    |> Result.list_to_result()
+    |> Result.map_err(fn decode_errors -> {:error, decode_errors} end) do
+      conf = config()
+      metadata_from_jwt = conf.jwt_fields
```

Comment:
isn't that a mapping between field names in the metadata and field names in the jwt?